packages/CakePHP/tests/Rector/MethodCall/RenameMethodCallBasedOnParameterRector/RenameMethodCallBasedOnParameterRectorTest.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace Rector\CakePHP\Tests\Rector\MethodCall\RenameMethodCallBasedOnParameterRector;
+
+use Rector\CakePHP\Rector\MethodCall\RenameMethodCallBasedOnParameterRector;
+use Rector\CakePHP\Tests\Rector\MethodCall\RenameMethodCallBasedOnParameterRector\Source\SomeModelType;
+use Rector\Testing\PHPUnit\AbstractRectorTestCase;
+
+final class RenameMethodCallBasedOnParameterRectorTest extends AbstractRectorTestCase
+{
+    /**
+     * @dataProvider provideDataForTest()
+     */
+    public function test(string $file): void
+    {
+        $this->doTestFile($file);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function provideDataForTest(): iterable
+    {
+        yield [__DIR__ . '/Fixture/fixture.php.inc'];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    protected function getRectorsWithConfiguration(): array
+    {
+        return [
+            RenameMethodCallBasedOnParameterRector::class => [
+                '$methodNamesByTypes' => [
+                    SomeModelType::class => [
+                        'getParam' => [
+                            'matchParameter' => 'paging',
+                            'replaceWith' => 'getAttribute',
+                        ],
+                        'withParam' => [
+                            'matchParameter' =>  'paging',
+                            'replaceWith' => 'withAttribute',
```

Comment:
Keys are better done in underscore, as it's yaml convention to separate from PHP method/property naming